ovos_config/base.py

```diff
@@ -0,0 +1,272 @@
+        translate_remote(config[module], v)
+
+
+class LocalConf(dict):
+    """Config dictionary from file."""
+    allow_overwrite = True
+
+    def __init__(self, path):
+        super().__init__(self)
+        self.path = path
+        if path:
+            self.load_local(path)
+
+    def _get_file_format(self, path=None):
+        """The config file format
+        supported file extensions:
+        - json (.json)
+        - commented json (.conf)
+        - yaml (.yaml/.yml)
+
+        returns "yaml" or "json"
+        """
+        path = path or self.path
+        if not path:
+            return "dict"
+        if path.endswith(".yml") or path.endswith(".yaml"):
+            return "yaml"
+        else:
+            return "json"
+
+    def load_local(self, path=None):
+        """Load local json file into self.
+
+        Args:
+            path (str): file to load
+        """
+        path = path or self.path
+        if not path:
+            LOG.error(f"in memory configuration, nothing to load")
+            return
+        if exists(path) and isfile(path):
+            try:
+                if self._get_file_format(path) == "yaml":
+                    with open(path) as f:
+                        config = yaml.safe_load(f)
+                else:
+                    config = load_commented_json(path)
+                for key in config:
+                    self.__setitem__(key, config[key])
+                LOG.debug(f"Configuration {path} loaded")
+            except Exception as e:
+                LOG.exception(f"Error loading configuration '{path}'")
+        else:
+            LOG.debug(f"Configuration '{path}' not defined, skipping")
+
+    def reload(self):
+        self.load_local(self.path)
+
+    def store(self, path=None):
+        """Cache the received settings locally.
+
+        The cache will be used if the remote is unreachable to load settings
+        that are as close to the user's as possible.
+        """
+        path = path or self.path
+        if not path:
+            LOG.error(f"in memory configuration, no save location")
+            return
+        if self._get_file_format(path) == "yaml":
+            with open(path, 'w') as f:
+                yaml.dump(dict(self), f, allow_unicode=True,
+                          default_flow_style=False, sort_keys=False)
+        else:
+            with open(path, 'w') as f:
+                json.dump(self, f, indent=2)
+
+    def merge(self, conf):
+        merge_dict(self, conf)
+
+
+class ReadOnlyConfig(LocalConf):
+    """ read only  """
+
+    def __init__(self, path, allow_overwrite=False):
+        super().__init__(path)
+        self.allow_overwrite = allow_overwrite
+
+    def reload(self):
+        old = self.allow_overwrite
+        self.allow_overwrite = True
+        super().reload()
+        self.allow_overwrite = old
+
+    def __setitem__(self, key, value):
+        if not self.allow_overwrite:
+            raise PermissionError
```

Comment:
Consider adding a string to this raised exception indicating this is a RO config and not a file system permissions error, etc